src/fs/dirlist.rs

```diff
@@ -10,25 +12,31 @@ pub struct JoshutoDirList {
     path: path::PathBuf,
     pub contents: Vec<JoshutoDirEntry>,
     pub metadata: JoshutoMetadata,
-    pub index: Option<usize>,
-    pub viewport_index: usize,
+    /// The cursor position in this dir list
+    index: Option<usize>,
+    /// The index in this dir list to start with when rendering the list
+    viewport_index: usize,
     _need_update: bool,
+    scroll_offset: usize,
```

Comment:
What do you think about putting `scroll_offset` somewhere else?
Ie. `UiContext`, or just leaving it in `DisplayOptions`
This will reduce the amount of memory used because we don't really need multiple copies of `scroll_offset` everywhere.

Reply:
Should be possible.
* `UiContext` does not feel like the right place for me. I thought about it as a place to store pre-calc UI information from the rendering process. On the other hand, that would be the cheapest solution.
* `DisplayOptions` is the place where it belongs, but that would mean that we need to pass DisplayOptions (or a clone) each time we call `set_index`.

I will think about it. ;)

Reply:
Done. Passing `DisplayOptions` (or a clone) to `set_index`.